pkg/lib/accountmanagement/service_authenticator.go

```diff
@@ -104,6 +117,18 @@ func (s *Service) ChangeSecondaryPassword(resolvedSession session.ResolvedSessio
 			OldPassword: input.OldPassword,
 			NewPassword: input.NewPassword,
 		})
+		if err != nil {
+			return err
+		}
+		// Changed password successfully
+		userID := resolvedSession.GetAuthenticationInfo().UserID
+		err = s.Events.DispatchEventOnCommit(&nonblocking.UserSettingsSecondaryPasswordChangedEventPayload{
+			UserRef: model.UserRef{
+				Meta: model.Meta{
+					ID: userID,
+				},
+			},
+		})
```

Comment:
Move the event dispatch to `s.changePassword`. To ensure every password change has a event.